### PR TITLE
Update dependencies

### LINKED ISSUE #13: Building test suite fails on Windows
README instructions for building are 
```
(mkdir -p build && cd build && cmake .. && make)
```
There is no additional information about restrictions to platform or dependencies, so my assumption is that it can run on any platform and will install any dependencies if not.

So bug is either broken code, or incomplete readme listing of dependencies/platforms.

This failed on Windows and Ubuntu Linux in same place
- windows 10 : https://gist.github.com/hamishwillee/63d73e88f35c8abdf05488a495579bd2
- Ubuntu Linux 16.04 - https://gist.github.com/hamishwillee/e11c1a430aad9b8db4cab8f013b6a12a

### LINKED ISSUE #25: Building test suite fails in a different way on Windows 10 (zlib build)
This _appears_ to be different from issue #13 of a similar name in that this build err on Win 10 occurs during the zlib build.  Teh google suggests this is a result of a dependency on the MSVC runtime library on a Cygwin-equipped box.

Should I simply install an MSVC runtime env and be done with it? I'm not sure which version.  TIA Howard

[edited to show errs exposed built within a cygwin env, orig post exhibited same error regarding missing _wopen but without benefit (?) of the cygwin environment]

```
[ 55%] Building C object CMakeFiles/zlib.dir/inftrees.o
[ 60%] Building C object CMakeFiles/zlib.dir/inffast.o
[ 65%] Building C object CMakeFiles/zlib.dir/trees.o
/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib/gzlib.c: In function ‘gz_open’:
/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib/gzlib.c:243:20: warning: implicit declaration of function ‘_wopen’; did you mean ‘fwopen’? [-Wimplicit-function-declaration]
         fd == -2 ? _wopen(path, oflag, 0666) :
                    ^~~~~~
                    fwopen
[ 70%] Building C object CMakeFiles/zlib.dir/uncompr.o
[ 75%] Building C object CMakeFiles/zlib.dir/zutil.o
[ 80%] Linking C static library libz.a
make[8]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
[ 80%] Built target zlib
make[8]: Entering directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[8]: Entering directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
Scanning dependencies of target example
make[8]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
Scanning dependencies of target minigzip
make[8]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[8]: Entering directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[8]: Entering directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
[ 85%] Building C object CMakeFiles/example.dir/test/example.o
[ 90%] Building C object CMakeFiles/minigzip.dir/test/minigzip.o
[ 95%] Linking C executable example.exe
[100%] Linking C executable minigzip.exe
libz.a(gzlib.o):gzlib.c:(.text+0x3e0): undefined reference to `_wopen'
libz.a(gzlib.o):gzlib.c:(.text+0x3e0): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_wopen'
libz.a(gzlib.o):gzlib.c:(.text+0x3e0): undefined reference to `_wopen'
libz.a(gzlib.o):gzlib.c:(.text+0x3e0): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_wopen'
collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
make[8]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[8]: *** [CMakeFiles/example.dir/build.make:96: example.exe] Error 1
make[8]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[8]: *** [CMakeFiles/minigzip.dir/build.make:96: minigzip.exe] Error 1
make[7]: *** [CMakeFiles/Makefile2:68: CMakeFiles/example.dir/all] Error 2
make[7]: *** Waiting for unfinished jobs....
make[7]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[7]: *** [CMakeFiles/Makefile2:142: CMakeFiles/minigzip.dir/all] Error 2
make[6]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib/zlib/src/zlib-build'
make[6]: *** [Makefile:139: all] Error 2
make[5]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib'
make[5]: *** [CMakeFiles/zlib.dir/build.make:111: zlib/src/zlib-stamp/zlib-build] Error 2
make[4]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib'
make[4]: *** [CMakeFiles/Makefile2:68: CMakeFiles/zlib.dir/all] Error 2
make[3]: Leaving directory '/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib'
make[3]: *** [Makefile:84: all] Error 2
CMake Error at third_party/cmake/build_target.cmake:54 (message):

  /cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/third_party/zlib
  failed to build!
Call Stack (most recent call first):
  third_party/CMakeLists.txt:14 (build_target)


-- Configuring incomplete, errors occurred!
See also "/cygdrive/c/Users/HowardGoldstein/Documents/mavlink-testing-suite-master/build/build_mavsdk/CMakeFiles/CMakeOutput.log".
make[2]: *** [CMakeFiles/third_party_mavsdk.dir/build.make:106: third_party_mavsdk-prefix/src/third_party_mavsdk-stamp/third_party_mavsdk-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:106: CMakeFiles/third_party_mavsdk.dir/all] Error 2
make: *** [Makefile:84: all] Error 2


```

### PR DESCRIPTION
**Pull request by julianoes**
See commits. Closes #13, closes #25.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -48,21 +48,21 @@ int main(int argc, char** argv)
...
+            mavsdk.subscribe_on_new_system(nullptr);
+            prom->set_value();
         } catch (const std::future_error& e) {
             // Ignore if prom is set multiple times which can happen even though we're
             // unsubscribing straightaway if the callback triggers two times in quick
             // succession.
         }
     });
 
+    auto system = mavsdk.systems()[0];
```
Comment (bkueng) [inline src/main.cpp:63]: Is this guaranteed to have 1 item?

**Review by bkueng — COMMENTED** [review]
I guess you forgot to update MAVSDK?

File: src/main.cpp (hunk 1 above)
Comment (julianoes) [reply src/main.cpp:63]: It should after the wait above, yes.

**Comment by julianoes** [comment]
> I guess you forgot to update MAVSDK?

Of course :man_facepalming:.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      run: sudo apt-get update && sudo apt-get install -y build-essential cmake clang-format
+    - name: Configure
+      run: cmake -j 2 -Bbuild -H.
+    - name: Build
+      run: cmake --build build -j 2
+    - name: Run clang-format
+      run: cmake --build build --target format
+
+  windows:
+    name: Windows
```
Comment (bkueng) [inline .github/workflows/main.yml:27]: :scream:

**Pull request merged**